src/core/helpers/util.ts

```diff
@@ -32,3 +34,34 @@ export function ensureBrowser(): void {
     );
   }
 }
+
+export async function readContract(
+  contract_address: string,
+  function_name: string,
+  function_params: any[],
+  output_params: string[],
+  caller_address: string
+) {
+  return await MainApi.instance().readContract({
+    contract_address,
+    function_name,
+    function_params,
+    output_params,
+    caller_address,
+  });
+}
+
+
+export async function writeContract(
+  contract_address: string,
+  function_name: string,
+  function_params: EncodedFunctionParameter[],
+  eth_amount: string,
+) {
+  return await MainApi.instance().createFunctionCallRequest({
```

Comment:
Wrap the FunctionCallRequest object (returned from MainAPI) inside the ContractCall object, which already contains the present() function used to show the widget.

```
return new ContractCallAction(
    await MainApi.instance().createFunctionCallRequest({
        ...
    })
);
```
Also add the return type of the outer `writeContract` to be set to `Promise<ContractCallAction>`